presto-spi/src/main/java/com/facebook/presto/spi/resourceGroups/SelectionContext.java

```diff
@@ -13,17 +13,28 @@
  */
 package com.facebook.presto.spi.resourceGroups;
 
+import java.util.Optional;
+
 import static java.util.Objects.requireNonNull;
 
 public final class SelectionContext<T>
 {
     private final ResourceGroupId resourceGroupId;
     private final T context;
+    private final Optional<ResourceGroupSelectionContext> resourceGroupSelectionContext;
```

Comment:
Can you better explain the decision to go for a separate class, rather than maybe just adding `dynamicFragmentId` here directly?

Reply:
I was thinking that we could add more properties here. Swapnil had some plans. I can change it to dynamicFragmentId. Another option was adding this to the ResourceGroupId directly.